tor/core/admin_commands.py

```diff
@@ -41,14 +42,74 @@ def process_override(reply, config):
         logging.info(
             'Starting validation override for post {}'
             ', approved by {}'.format(
-                parents_parent.fullname, reply.author.name
+                parents_parent.fullname,
+                reply.author.name
             )
         )
         process_done(
             parents_parent, config, override=True
         )
 
 
+def process_blacklist(reply, config):
+    """
+    This is used to basically "shadow-ban" people from the bot.
+    Format is:
+    Subject: !blacklist
+    body: <username1>\n<username2>...
+    :param reply: the comment reply object from the inbox
+    :param config: the global config object
+    :return: None
+    """
+
+    if not from_moderator(reply, config):
+        reply.reply(_(random.choice(config.no_gifs)))
+        logging.info(
+            '{} just tried to blacklist. Get your own bot!'
+            ''.format(reply.author.name)
+        )
+        return
+
+    usernames = reply.body.splitlines()
+    results = ""
+    failed = []
+    successes = []
+    already_added = []
+
+    for username in usernames:
+        if username in config.tor_mods:
+            results += f"{username} is a mod! Don't blacklist mods!\n"
+            failed.append(username)
+            continue
+
+        try:
+            config.r.redditor(username)
+        except RedditClientException:
+            results += f"{username} isn't a valid user\n"
+            failed.append(username)
+            continue
+
+        already_added = config.redis.sadd('blacklist', username)
+        if already_added == 0:
+            results += f"{username} is already blacklisted, ya fool!\n"
+            already_added.append(username)
+            continue
+
+        results += f"{username} is now blacklisted\n"
+        successes.append(username)
+
+        reply.reply(results)
+
+        logging.info(
+            "Blacklist: {failed} failed, {success} succeeded, {ignored} were "
+            "already blacklisted".format(
+                failed=repr(failed),
+                success=repr(successes),
+                ignored=repr(already_added)
+            )
```

Comment:
Normally, I'd comment on the intermixed f-strings and `str.format()` style of string interpolation in the codebase, but it sounds like @itsthejoker is warming up to f-strings. I'm just a hold-out of overly traditional methods, I suppose. Looks good to me!